apis/acm/v1beta1/certificate_types.go

```diff
@@ -124,6 +125,24 @@ type CertificateParameters struct {
 	// +immutable
 	DomainValidationOptions []*DomainValidationOption `json:"domainValidationOptions,omitempty"`
 
+	// Specifies the algorithm of the public and private key pair that your certificate
+	// uses to encrypt data. RSA is the default key algorithm for ACM certificates.
+	// Elliptic Curve Digital Signature Algorithm (ECDSA) keys are smaller, offering
+	// security comparable to RSA keys but with greater computing efficiency. However,
+	// ECDSA is not supported by all network clients. Some AWS services may require RSA
+	// keys, or only support ECDSA keys of a particular size, while others allow the
+	// use of either RSA and ECDSA keys to ensure that compatibility is not broken.
+	// Check the requirements for the AWS service where you plan to deploy your
+	// certificate.
+	// Default: RSA_2048
+	// Note: ACM can request the issue of new certificates using the following algorithms:
+	// "RSA_2048", "EC_prime256v1" and "EC_secp384r1".
+	// The remaining algorithms are supported only for imported certificates.
+	// See also AWS docs: https://docs.aws.amazon.com/acm/latest/userguide/acm-certificate.html#algorithms.title
+	// +optional
+	// +immutable
+	KeyAlgorithm types.KeyAlgorithm `json:"keyAlgorithm,omitempty"`
```

Comment:
I think it would be better to have a dedicated type `KeyAlgorithm` in `apis` to not be dependend on the AWS SDK.

I also think it should be a pointer because it is optional?

Reply:
Type: I was inspired by `CertificateAuthority` https://github.com/crossplane-contrib/provider-aws/blob/1712285bd6460d056c3157a01344328d6d0bb648/apis/acmpca/v1beta1/certificateauthority_types.go#L90

This way we do not have to keep a dedicated type uptodate over sdk-upgrades?

Pointer: hmm probably got confused by inconsistency :https://github.com/crossplane-contrib/provider-aws/blob/1712285bd6460d056c3157a01344328d6d0bb648/apis/acm/v1beta1/certificate_types.go#L166

Reply:
> This way we do not have to keep a dedicated type uptodate over sdk-upgrades?

Unfortunately we do, because it is only an alias for string. Kubebuilder does not detect enums that way. Also it is not good design to have the API types depend on what is used in the implementation. They should be separated.

Reply:
Ok. Thanks for the explanations.
Updated the first commit `feat(certificate): add key algorithm` with the suggestions.